### ISSUE TITLE
Unable to create application from private SSH repository: key is unknown

### ISSUE THREAD

**Issue by basvandenbrink** [argoproj/argo-cd#1743]
**Describe the bug**
I cannot create an application based on a private Git repository using SSH.

**To Reproduce**
1. Modify Argo CD deployment according to [this documentation](https://argoproj.github.io/argo-cd/user-guide/private-repositories/#unknown-ssh-hosts).
2. Create a Secret containing the private SSH key and configure the repository using a ConfigMap as [documented here](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#repositories).
3. Verify that Argo CD is connected successfully using the web gui.
4. Create an application from this repository using the web gui or cli.

**Expected behavior**
I expected that the application is created successfully and that Argo CD deploys this application on the cluster.

**Version**

```shell
argocd: v1.0.1+5fe1447.dirty
  BuildDate: 2019-05-28T17:26:35Z
  GitCommit: 5fe1447b722716649143c63f9fc054886d5b111c
  GitTreeState: dirty
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.0.1+5fe1447.dirty
  BuildDate: 2019-05-28T17:27:38Z
  GitCommit: 5fe1447b722716649143c63f9fc054886d5b111c
  GitTreeState: dirty
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: 0.13.1
```

**Logs**

```
Unable to create application: application spec is invalid: InvalidSpecError: Unable to determine app source type: rpc error: code = Unknown desc = ssh: handshake failed: knownhosts: key is unknown
```

**Comment by alexec** [comment]
1. Can you please tell me what tool you're using. Helm? Kustomize? Other?
2. Can you please provide a link to to the Git repository?

**Comment by alexec** [comment]
Maybe fixed by #827 if Kustomize.

**Comment by basvandenbrink** [comment]
I am using a Helm charts that is stored on the repository.

Thanks for your reaction.

I looked at the issue you mentioned and executing `ssh-keyscan <url> > ~/.ssh/known_hosts` on the repo server solved the issue.

Another thing: [the example shown here](https://github.com/argoproj/argo-cd/tree/master/examples/known-hosts) states that known_hosts file has to be hashed (`ssh-keygen -Hf known_hosts`). I have tried to copy `/etc/ssh/ssh_known_hosts` to `~/.ssh/known_hosts`, but that doesn't work, since `~/.ssh/known_hosts` may not be hashed for some reason.

Looking forward to v1.1.0. I hope it is then fixed. Is there also going to be a native option to add known_hosts entries?

**Comment by basvandenbrink** [comment]
@alexec Is the issue also fixed for Helm charts stored on private Git repositories with SSH access?

**Comment by alexec** [comment]
Hi. In hindsight, I don't think it is fixed by the fix I mentioned unless you use the insecure option. You would not want to do that for prod instances of Argo CD. 

@jessesuen I'd like us to consider adding keys on start-up for your app. Thoughts?

**Comment by alexec** [comment]
See #1514, #1513

**Comment by basvandenbrink** [comment]
Closing, because the mentioned issues also cover this problem. Would be resolved when pull request #1807 is merged.
